modules/check_in/spec/request/v2/sessions_request_spec.rb

```diff
@@ -79,10 +76,105 @@
   end
 
   describe 'POST `create`' do
-    it 'returns not implemented' do
-      post check_in.v2_sessions_path, params: {}
+    context 'when invalid params' do
+      let(:invalid_uuid) { 'invalid_uuid' }
+      let(:resp) do
+        {
+          'error' => true,
+          'message' => 'Invalid uuid, last4 or last name!'
+        }
+      end
+      let(:session_params) do
+        {
+          params: {
+            session: {
+              uuid: invalid_uuid,
+              last4: '555',
+              last_name: ''
+            }
+          }
+        }
+      end
+
+      it 'returns an error response' do
+        post '/check_in/v2/sessions', session_params
+
+        expect(response.status).to eq(400)
+        expect(JSON.parse(response.body)).to eq(resp)
+      end
+    end
+
+    context 'when authorized' do
+      let(:uuid) { Faker::Internet.uuid }
+      let(:resp) do
+        {
+          'permissions' => 'read.full',
+          'status' => 'success',
+          'uuid' => uuid
+        }
+      end
+      let(:session_params) do
+        {
+          params: {
+            session: {
+              uuid: uuid,
+              last4: '5555',
+              last_name: 'Johnson'
+            }
+          }
+        }
+      end
+      let(:key) { "check_in_lorota_v2_#{uuid}_read.full" }
+
+      it 'returns an error response' do
+        allow_any_instance_of(CheckIn::V2::Session).to receive(:redis_session_prefix).and_return('check_in_lorota_v2')
+        allow_any_instance_of(CheckIn::V2::Session).to receive(:jwt).and_return('jwt-123-1bc')
+
+        Rails.cache.write(key, 'jwt-123-1bc', namespace: 'check-in-lorota-v2-cache')
+
+        post '/check_in/v2/sessions', session_params
+
+        expect(response.status).to eq(200)
+        expect(JSON.parse(response.body)).to eq(resp)
+      end
+    end
 
-      expect(response.status).to eq(501)
+    context 'when unauthorized' do
```

Comment:
Is this "unauthorized" flow, since the response both from this service and LoROTA are success responses? Or is it just worded incorrectly?

Reply:
Good spot. Something definitely went wrong there. I'll take a look 👍